presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcMetadata.java

```diff
@@ -120,7 +120,7 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
     public Map<SchemaTableName, List<ColumnMetadata>> listTableColumns(ConnectorSession session, SchemaTablePrefix prefix)
     {
         ImmutableMap.Builder<SchemaTableName, List<ColumnMetadata>> columns = ImmutableMap.builder();
-        for (SchemaTableName tableName : listTables(session, prefix.getSchemaName())) {
+        for (SchemaTableName tableName : jdbcClient.getTableNames(prefix.getSchemaName(), prefix.getTableName())) {
```

Comment:
`SchemaTablePrefix` is perhaps confusingly named and should have documentation. It has three possible states:

1. Neither schema nor table
2. Schema but not table
3. Schema and table

If schema or table is set, it is a full name, not prefix or search pattern.

Thus, we can do something like this here:
```java
List<SchemaTableName> tables;
if (prefix.getTableName() != null) {
    tables = ImmutableList.of(new SchemaTableName(prefix.getSchemaName(), prefix.getTableName()));
}
else {
    tables = listTables(session, prefix.getSchemaName());
}